presto-docs/src/main/sphinx/functions/datetime.rst

```diff
@@ -100,26 +101,28 @@ Date and Time Functions
         SELECT from_iso8601_date('2020-123');
         -- 2020-05-02
 
-.. function:: at_timezone(timestamp, zone) -> timestamp with time zone
+.. function:: at_timezone(timestamp(p), zone) -> timestamp(p) with time zone
```

Comment:
The `timestamp` argument should not change since this is an argument name, not a type. The only change should be adding precision to the return type. The description should not change.
```
.. function:: at_timezone(timestamp, zone) -> timestamp(p) with time zone

Reply:
So at this stage it doesnt seem clear if this supports the data type "TIMESTAMP WITH TIME ZONE" or "TIMESTAMP WITHOUT TIME ZONE" or both ... what is actually supported and how do we best communicate that. I think it only supports timestamp with time zone .. but I am not sure. E.g. it could do a conversion ... @martint ? @electrum ?

Reply:
`at_timezone` applies to `timestamp with time zone`, not to `timestamp`, so the current explanation is a little confusing.

Reply:
Great.. so you are confirming that it only applies to timestamp with time zone .. so we should rename the input parameter to make that clear .. any preference for name timestampt_with_time_zone or so @electrum ?

Reply:
Discussed with @electrum and we will leave this simple for now. As a follow up we might need to document the implicit type coercion how it applies to function and specifically also timestamp with and without time zone..